Move check if useStorageBuffer needs to be set

FoggyMist opened the pull request:
...From TParseContext used only by GLSL, to TParseContextBase inherited by both GLSL and HLSL paths. It caused compilations from HLSL to SPIR-V 1.3+ to use BufferBlock decoration which is no longer valid.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/glslang?pullRequest=3121) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/glslang?pullRequest=3121) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/glslang?pullRequest=3121) it.</sub>

FoggyMist commented:
As mentioned in https://github.com/KhronosGroup/glslang/issues/2411 some HLSL inputs are producing invalid SPIR-V output. When it happens it can be easily validated that output SPIR-V is using outdated operand of Decorate. BufferBlock has been deprecated in SPIR-V 1.3 and removed in newer versions.

My debug session went like this:

1. I found that there is only one part of the code that produces BufferBlock token during translation to SPIR-V. It is `TranslateBlockDecoration` function from `glslang/SPIRV/GlslangToSpv.cpp:378`.
2. This function is being called in only one place `glslang/SPIRV/GlslangToSpv.cpp:4842`. It is being called when compiling both HLSL and GLSL, but with different `useStorageBuffer`.
3. That flag comes from `glslangIntermediate` member that is being initialized by ParseContext class.
4. There is switch with `#ifdef` that picks right concrete implementation of ParseContext, depending on shader input (HLSL vs GLSL).
5. It turns out that `TParseContext` (which is part of GLSL path) checks if `--target-env` is SPIR-V 1.3 or newer inside constructor. But there is no such check in `HlslParseContext`'s constructor (HLSL path).
6. I moved check to common base class - `TParseContextBase`. It worked, so I regenerated test files and submitted this PR.

arcady-lunarg reviewed (approved):
Looks good.
